renderer/src/GLRenderer.cpp

```diff
@@ -312,7 +315,9 @@ void GLRendererImplementation::render_loop() {
         BOOST_LOG_TRIVIAL(error) << "error hex code " << std::hex << error;
     }
 
-    glXSwapBuffers(rendering_context_->x_display, rendering_context_->gl_x_window);
+    glXSwapBuffers(rendering_context_->get_display(), rendering_context_->get_drawable());
+    auto end = std::chrono::steady_clock::now();
+    rendering_context_->set_current_time(std::chrono::duration_cast<std::chrono::microseconds>(end - start).count());
```

Comment:
Maybe it's better to pass actual duration into the method, and invoke `count()` inside.